src/main.rs

```diff
@@ -67,7 +67,13 @@ fn validate_simulation() {
 
 /// Simulate the rabbit population
 fn simulate(starting_rabbits: i128) -> i32 {
-    todo!()
+    if starting_rabbits == 1 {
+        return 0;
+    } else if starting_rabbits % 2 != 0 {
+        return 1 + simulate(starting_rabbits * 3 + 1);
+    } else {
+        return 1 + simulate(starting_rabbits / 2);
+    }
```

Comment:
Nicely done on the recursion! We can actually get a bit more interesting here; if we don't return inside the if statement, then the whole if statement will evaluate to a single number. This is what I mean:

```suggestion
    if starting_rabbits == 1 {
        0
    } else if starting_rabbits % 2 != 0 {
        1 + simulate(starting_rabbits * 3 + 1)
    } else {
        1 + simulate(starting_rabbits / 2)
    }
```